webapp/javascript/components/FlameGraph/enchancers.tsx

```diff
@@ -0,0 +1,72 @@
+import { connect } from 'react-redux';
+import { RootState } from '@pyroscope/redux/store';
+import { selectUIState } from '@pyroscope/redux/reducers/views';
+
+import { setUIValue } from '@pyroscope/redux/actions';
+
+type IPersistedValueSetters<Type> = {
+  [Property in keyof Type as `set${Capitalize<string & Property>}`]: (
+    v
+  ) => void;
+};
+
+export type IPersistedValue<T> = IPersistedValueSetters<T> & T;
+
+interface ICurrentUIView {
+  view: string;
+}
+
+const capitalize = (s) => s.charAt(0).toUpperCase() + s.slice(1);
+
+const withUpdateableUIValue = (valueName, savePath, defaultValue = undefined) =>
+  connect(
+    (state: RootState) => ({
+      [valueName]: selectUIState(state)(savePath, defaultValue),
+    }),
+    (dispatch) => ({
+      [`set${capitalize(valueName)}`]: (value) => {
+        dispatch(setUIValue(savePath, value));
+      },
+    })
+  );
+
```

Comment:
Do we need any of this code? Except for `capitalize` I don't see the rest being used?

Reply:
No we don't. Just as an example of a single value being saved to state

Reply:
Removed unused code